crates/fluvio-version-manager/src/command/itself/install.rs

```diff
@@ -50,24 +40,46 @@ impl SelfInstallOpt {
 
     /// Creates the `~/.fvm` directory and copies the current binary to this
     /// directory.
+    ///
+    /// # Usage of `create_dir` over `create_dir_all`
+    ///
+    /// Given that on updates the directories may be present, to avoid failing
+    /// on `create_dir`, `create_dir_all` is used instead.
+    ///
+    /// Something similar happens on `mkdir` command, even though underlaying
+    /// syscalls may differ.
+    ///
+    /// Consider the existent directory `~/.fvm/versions`, executing `create_dir`
+    /// will fail with error:
+    ///
+    /// ```ignore
+    /// ~/.fvm/versions: File exists
+    /// ```
+    ///
+    /// Instead by doing `create_dir_all` the error will not happen.
+    ///
+    /// ```ignore
+    /// mkdir -p ~/.fvm/versions
+    /// ```
+    ///
     fn install_fvm(&self) -> Result<PathBuf> {
         // Creates the directory `~/.fvm` if doesn't exists
         let fvm_dir = fvm_workdir_path()?;
 
-        if !fvm_dir.exists() {
-            create_dir(&fvm_dir)?;
-            tracing::debug!(?fvm_dir, "Created FVM home directory with success");
-        }
-
-        // Attempts to create the binary crate
-        let fvm_binary_dir = fvm_dir.join("bin");
-        create_dir(&fvm_binary_dir)?;
-        tracing::debug!(?fvm_binary_dir, "Created FVM bin directory with success");
+        // Creates the binaries directory
+        let bin_dir = fvm_dir.join("bin");
+        create_dir_all(bin_dir)?;
```

Comment:
Instead of checking for directories before creating:

```rust
 if !fvm_dir.exists() {
   create_dir(&fvm_dir)?;
   tracing::debug!(?fvm_dir, "Created FVM home directory with success");
 }
```

We can use `create_dir_all` which not only create parent dirs in a single execution
but it also allows for no-op on already existent dirs.